jcommune-model/src/test/java/org/jtalks/jcommune/model/dao/hibernate/UserHibernateDaoTest.java

```diff
@@ -685,37 +690,45 @@ public void testFindUser_IfHeHasNoGroups() {
 
     @Test(invocationCount = 5)
     public void testDontFindUser_IfHeIsInGroup() {
-        final String keyWord = "keyword";
-        Group group1 = createGroup("testGroup1");
         Random rnd = new Random();
+        final String keyWord = randomAlphanumeric(7);;
+        List<Group> grpList;
+        JCUser user;
         if (rnd.nextBoolean()) {
-            createUserWithGroup(keyWord, "mail@mail.ru", group1);
+            user = createUserWithGroup(keyWord, "mail@mail.ru", createGroup("testGroup0"));
         } else {
-            createUserWithGroup("username", keyWord + "@mail.ru", group1);
+            user = createUserWithGroup("username", keyWord + "@mail.ru", createGroup("testGroup0"));
+        }
+        grpList = addRandomGroupsToUser(user);
+        for (Group group : grpList) {
```

Comment:
From my point of view If we are checking for all added group we provide better test coverage.